vault_service/views/user.py

```diff
@@ -195,4 +197,451 @@ def store_data():
+    # of the setup from the dict (e.g. active, frequency, etc.)
+
+    # TODO check rate limit
+
+    # structure in vault:
+    # "myADS": [{"name": user-supplied name, "qid": QID from query table (type="query") or ID from myads table
+    # (type="template"), "type":  "query" or "template", "active": true/false, "frequency": "daily" or "weekly",
+    # "stateful": true/false}]
+
+    output = []
+    with current_app.session_scope() as session:
+        setups = session.query(MyADS).filter_by(user_id=user_id).filter_by(active=True).order_by(MyADS.id.asc()).all()
+        if not setups:
+            return '{}', 404
+        for s in setups:
+            o = {'id': s.id,
+                 'name': s.name,
+                 'type': s.type,
+                 'active': s.active,
+                 'stateful': s.stateful,
+                 'frequency': s.frequency,
+                 'template': s.template,
+                 'classes': s.classes,
+                 'data': s.data,
+                 'created': s.created.isoformat(),
+                 'updated': s.updated.isoformat()}
+
+            if s.type == 'query':
+                try:
+                    q = session.query(Query).filter_by(id=s.query_id).one()
+                    qid = q.qid
+                except ormexc.NoResultFound:
+                    qid = None
+            else:
+                qid = None
+            o['qid'] = qid
+
+            output.append(o)
+
+    return json.dumps(output), 200
+
+
+@advertise(scopes=['ads-consumer:myads'], rate_limit = [1000, 3600*24])
+@bp.route('/myads-users/<iso_datestring>', methods=['GET'])
+def export(iso_datestring):
+    '''
+    Get the latest changes (as recorded in users table)
+        The optional argument latest_point is RFC3339, ie. '2008-09-03T20:56:35.450686Z'
+    '''
+    # TODO check rate limit
+    # TODO need to add a created/update field to users table
+
+    # inspired by orcid-service endpoint of same endpoint, checks the users table for profiles w/ a myADS setup that
+    # have been updated since a given date/time; returns these profiles to be added to the myADS processing
+
+    latest = parser.parse(iso_datestring) # ISO 8601
+    output = []
+    with current_app.session_scope() as session:
+        setups = session.query(MyADS).filter(MyADS.updated > latest).order_by(MyADS.updated.asc()).all()
+        for s in setups:
+            output.append(s.user_id)
+
+    output = list(set(output))
+    return json.dumps({'users': output}), 200
+
+
+@advertise(scopes=['import-myads'], rate_limit=[1000, 3600*24])
+@bp.route('/myads-import', methods=['GET'])
+def import_myads():
+    '''
+
+    :return:
+    '''
+    try:
+        payload, headers = check_request(request)
+    except Exception as e:
+        return json.dumps({'msg': e.message or e.description}), 400
+
+    user_id = int(headers['X-Adsws-Uid'])
+
+    if user_id == 0:
```

Comment:
Hard coded `0` that should be `current_app.config['BOOTSTRAP_USER_ID']`. Also, if we keep the anonymous user without this scope, it won't be able to request it so this check is not strictly necessary (although no huge impact if we leave it too).